src/app/utilities/AnnotationUtilities.ts

```diff
@@ -221,6 +221,58 @@ const SearchNestedLevels = (level: Dict | Dict[], nestingLevels: string[], targe
     }
 }
 
+/* Function for formatting the property path based upon the base model and its typings, indexes are assigned to the first parent class that represents an array */
+const FormatTargetPropertyPath = (path: string, targetType: string = 'DigitalSpecimen', index?: number) => {
+    let baseModel: Dict;
+    let formattedPath: string = '';
+
+        /* Prepare base model */
+    if (targetType === 'DigitalMedia') {
+        baseModel = DigitalEntity.properties;
+    } else {
+        baseModel = DigitalSpecimen.properties;
+    }
+
+    /* Function to check if type of property is array */
+    const CheckArrayType = (schema: Dict, level: string) => {
+        let formattedLevel = level;
+
+        if (schema[level].type === 'array') {
+            formattedLevel = `${level}[${index}]`;
+        }
+
+        return formattedLevel;
+    }
+
+    /* Function to concat level to field, add dot if index is greater than zero */
+    const ConcatToField = (level: string, index: number) => {
+        if (index > 0) {
+            formattedPath = formattedPath.concat(`.${level}`);
+        } else {
+            formattedPath = formattedPath.concat(level);
+        }
+    }
+
+    const levels = path.replace(`${targetType}.`, '').split('.');
```

Comment:
I think you're replacing/removing a `.` then splitting on that character. Is that intentional?

Reply:
Ah yes, I can understand the confusion, this is to make sure that the top level component of a JSON path is removed.
Top level meaning 'DigitalSpecimen' or 'DigitalEntity' for example.
The targetType variable represents that top level, so I am clearing 'DigitalSpecimen' plus a '.' (dot) from the path, leaving the dot out would result in something like this: '.occurrences.' with the dot still in place at the start